etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/UpdateReceivingFacilityWithOrderingFacilityIdentifier.java

```diff
@@ -0,0 +1,33 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.custom;
+
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.CustomFhirTransformation;
+import gov.hhs.cdc.trustedintermediary.external.hapi.HapiHelper;
+import java.util.Map;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.DiagnosticReport;
+import org.hl7.fhir.r4.model.ServiceRequest;
+
+/**
+ * Updates the receiving facility (MSH-6) to value in Ordering Facility Name's Organization
+ * Identifier (ORC-21.10).
+ */
+public class UpdateReceivingFacilityWithOrderingFacilityIdentifier
+        implements CustomFhirTransformation {
+
+    @Override
+    public void transform(FhirResource<?> resource, Map<String, String> args) {
+        Bundle bundle = (Bundle) resource.getUnderlyingResource();
+        DiagnosticReport diagnosticReport = HapiHelper.getDiagnosticReport(bundle);
+        if (diagnosticReport == null) {
+            return;
+        }
+        ServiceRequest serviceRequest = HapiHelper.getBasedOnServiceRequest(diagnosticReport);
```

Comment:
This method name is confusing, what are we getting based on the service request?

Reply:
The `DiagnosticReport` has a `BasedOn` property that has a `ServiceRequest` in it. Do you have any suggestions how what to rename it to?

Reply:
I think a name like `getServiceRequest` would do the job, and just have the parameter be of type `IBase`

Reply:
Renamed